hms-app/src/components/HackathonSelectDropdown.tsx

```diff
@@ -27,38 +31,63 @@ export default function HackathonSelectDropdown(setHackathonID: Props) {
     )
   }
 
-  const hackathonMap = hackathonList.map((hackathon, index) => ({
-    value: hackathon.id,
-    label:
-      hackathon.title +
-      ' ' +
-      new Date(hackathon.startDate).toLocaleDateString() +
-      '-' +
-      new Date(hackathon.endDate).toLocaleDateString(),
-  }))
+  const futureAndPresentHackathons = hackathonList.filter((hackathon) => {
+    return new Date(hackathon.endDate) > today
+  })
 
-  const selectChange = (value: string) => {
-    setSelectedHackweek(value)
+  const pastHackathons = hackathonList.filter((hackathon) => {
+    return new Date(hackathon.endDate) < today
+  })
+
+  function getHackatonsForContext(): HackathonPreview[] {
+    switch (context) {
+      case 'archive':
+        return pastHackathons
+      case 'idea-portal':
+        return futureAndPresentHackathons
+    }
+    return hackathonList
+  }
+
+  function mapHackathonToSelectItem(hackathon: HackathonPreview): SelectItem {
+    return {
+      value: hackathon.id,
+      label:
+        hackathon.title +
+        ' ' +
+        new Date(hackathon.startDate).toLocaleString(undefined, {
+          year: 'numeric',
+          month: '2-digit',
+          day: '2-digit',
+        }) +
+        '-' +
+        new Date(hackathon.endDate).toLocaleString(undefined, {
+          year: 'numeric',
+          month: '2-digit',
+          day: '2-digit',
+        }),
+    }
   }
 
+  const hackathonMap = getHackatonsForContext().map((hackathon) =>
```

Comment:
would you ever call `getHackathonsForContext()` and not immediately call this map function on it? if not then you could probably just have `getHackathonsForContext()` also do the map